application/controllers/admin/Quotas.php

```diff
@@ -271,13 +297,17 @@ public function index(int $iSurveyId, bool $quickreport = false)
      */
     public function insertquotaanswer(int $iSurveyId)
     {
+        $this->requirePostRequest();
+
         $iSurveyId = sanitize_int($iSurveyId);
-        $this->checkPermissions($iSurveyId, 'update');
+        $qid = Yii::app()->request->getPost('quota_qid');
+        $quota_id = Yii::app()->request->getPost('quota_id');
+        $this->checkPermissions($iSurveyId, 'update', $quota_id);
```

Comment:
If I am not wrong, usually the check permission is done using the surveyId from the quota model.

Even more, if we have the quota model, why do we need surveyId parameter?
Maybe out of the scope of this fix, but at least we could stop using it and use the one from the quota model, right?

Reply:
If we need a function for checking a quota belongs to a survey, we could do it on a new function.

Reply:
>  Maybe out of the scope of this fix, but at least we could stop using it and use the one from the quota model, right?

Clearly out of scope : valid for Question, Answer, Questioni10n, Answeri10n etc …

Then : if we construct something must be in trait

Reply:
> If we need a function for checking a quota belongs to a survey, we could do it on a new function.

Same for ALL other model related to Survey : out of scope.

Reply:
You are checking the quota belongs to a survey on a check permission function. I am just saying, move that part to a new function.

Reply:
> You are checking the quota belongs to a survey on a check permission function. I am just saying, move that part to a new function.

the checkPermission function is a private function … i just rename it then.

Reply:
Added

```
    /**
     * Check validity of a quotaid against a survey id
     * @param integer $iSurveyId
     * @param integer $quotaid
     * throw Exception
     * @return void
     */
    private function checkValidQuotaId($surveyId, $quotaId)
    {
        $model = Quota::model()->find(
            "id = :quotaid AND sid = :surveyid",
            [':quotaid' => $quotaId, ':surveyid' => $surveyId]
        );
        if (empty($model)) {
            throw new CHttpException(403, gT("Bad quota id with this survey id."));
        }

    }
```